### PR TITLE
fix and add tests to parseImageTag function

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

issue from slack: https://astronomerteam.slack.com/archives/C02NZELT2/p1606179099323300

### CODE REVIEW

File: src/lib/deployments/config/index.unit.test.js [hunk 1]
```diff
@@ -1331,3 +1332,32 @@ describe("containerUidGid", () => {
...
+
+describe("parseImageTag", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
```
Comment (ashb) [inline src/lib/deployments/config/index.unit.test.js:1344]:
```suggestion
```

No mocks in this test to reset.

**Review by ashb — CHANGES_REQUESTED** [review]
We should also add a test of a "deployment-style" image tag to make sure that continues to work too.

**Comment by andriisoldatenko** [comment]
> We should also add a test of a "deployment-style" image tag to make sure that continues to work too.

yeah let me find few good examples in staging cluster.

**Comment by andriisoldatenko** [comment]
@ashb JFYI
```
group(kube_pod_container_info{pod=~".*[A-Za-z0-9-].*scheduler-.*[A-Za-z0-9-].*",container="scheduler"}) by (image)
```

```

{image="quay.io/astronomer/ap-airflow-dev:2.0.0-buster-onbuild-22919"} | 1
-- | --
{image="registry.staging.astronomer.io/arithmetic-fusion-3027/airflow:deploy-3"} | 1
{image="astronomerinc/ap-airflow:1.10.10-5-alpine3.10-onbuild"} | 1
{image="registry.staging.astronomer.io/popular-sun-9511/airflow:deploy-1"} | 1
{image="registry.staging.astronomer.io/meteoroidal-gravitation-1535/airflow:deploy-1"} | 1
{image="astronomerinc/ap-airflow:1.10.7-alpine3.10-onbuild"} | 1
{image="astronomerinc/ap-airflow:1.10.7-7-buster-onbuild"} | 1
{image="registry.staging.astronomer.io/astral-pulsar-8100/airflow:deploy-2"} | 1
{image="astronomerinc/ap-airflow:1.10.7-7-alpine3.10-onbuild"} | 1
{image="registry.staging.astronomer.io/glowing-wormhole-9119/airflow:deploy-1"} | 1
{image="registry.staging.astronomer.io/asteroidal-eccentricity-4447/airflow:deploy-15"} | 1
{image="registry.staging.astronomer.io/frigid-declination-9223/airflow:deploy-1"} | 1
{image="astronomerinc/ap-airflow:1.10.7-15-alpine3.10-onbuild"} | 1
{image="quay.io/astronomer/ap-airflow-dev:1.10.12-buster-onbuild"} | 1
{image="quay.io/astronomer/ap-airflow:1.10.7-buster"} | 1
{image="registry.staging.astronomer.io/ancient-declination-1784/airflow:deploy-1"} | 1
{image="quay.io/astronomer/ap-airflow-dev:1.10.10-buster"} | 1
{image="astronomerinc/ap-airflow:1.10.12-1-alpine3.10-onbuild"} | 1
{image="registry.staging.astronomer.io/physical-elliptical-1957/airflow:deploy-1"} | 1
{image="registry.staging.astronomer.io/interstellar-comet-6668/airflow:deploy-1"} | 1
{image="dimberman/helm-test-image:0.2.3"} | 1
{image="registry.staging.astronomer.io/glowing-star-3628/airflow:deploy-1"}

```

**Review by ashb — APPROVED** [review]

**Review by adam2k — APPROVED** [review]

**Pull request merged**